syntax = "proto3";

package metatrader;

service AccountService {

    // Gets static account information.
    rpc GetAccountInfo (Void) returns (AccounInfo);

    // Gets dynamic account status.
    rpc GetAccountStatus (Void) returns (AccountStatus);
}

message AccounInfo {
    int64 id = 1;
    string broker = 2;
    string currency = 3;
    int32 digits = 4;
}

message AccountStatus {
    float balance = 1;
    float equity = 2;
    float usedMargin = 3;
    float freeMargin = 4;
}

message Void {}
